harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -403,6 +397,12 @@ def _train_for_step(
             num_inputs *= hvd.size()
         metrics = det.util.make_metrics(num_inputs, per_batch_metrics)
 
+        # XXX: wait, it's not really right to apply the reducers to the the per_batch_metrics
```

Comment:
When we average training metrics, if the per-batch metrics from two machines were:
```
[1, 2, 3, 4]
[5, 6, 7, 8]
```
Rather than return something like:
```
[
  [1, 5],
  [2, 6],
  [3, 7],
  [4, 8],
]
```
or
```
[
  [1, 2, 3, 4],
  [5, 6, 7, 8]
]
```
what we return is:
```
[3, 4, 5, 6]
```
Which is only valid when the metrics are are averageable, and it's way harder and I think less helpful to the user.

I don't propose that we change it right now in any case, but I put that comment here to start a discussion, primarily around "should pytorch custom reducers interact with the `average_training_metrics` setting and that codepath or not".  My answer is "no it is independent" and I wanted to make sure that other people agreed.